src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+ *   ui = "Drupal\tmgmt_thebigword\ThebigwordTranslatorUi"
+ * )
+ */
+class ThebigwordTranslator extends TranslatorPluginBase implements ContainerFactoryPluginInterface, ContinuousTranslatorInterface, RemoteTranslatorInterface {
+
+  /**
+   * Translation service URL.
+   */
+  const PRODUCTION_URL = 'http://uat-integration.thebigword.com/drupal/api';
+
+  /**
+   * Translation service API version.
+   *
+   * @var string
+   */
+  const API_VERSION = '1';
+
+  /**
+   * The translator.
+   *
+   * @var TranslatorInterface
+   */
+  private $translator;
+
+  /**
+   * Sets a Translator.
+   *
+   * @param \Drupal\tmgmt\TranslatorInterface $translator
+   *   The translator to set.
+   */
+  public function setTranslator(TranslatorInterface $translator) {
+    if (!isset($this->translator)) {
+      $this->translator = $translator;
+    }
+  }
+
+  /**
+   * Guzzle HTTP client.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $client;
+
+  /**
+   * List of supported languages by Thebigword.
+   *
+   * @var array
+   */
+  protected $supportedRemoteLanguages = array();
+
+  /**
+   * Constructs a ThebigwordTranslator object.
+   *
+   * @param \GuzzleHttp\ClientInterface $client
+   *   The Guzzle HTTP client.
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param array $plugin_definition
+   *   The plugin implementation definition.
+   */
+  public function __construct(ClientInterface $client, array $configuration, $plugin_id, array $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->client = $client;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $container->get('http_client'),
+      $configuration,
+      $plugin_id,
+      $plugin_definition
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function requestTranslation(JobInterface $job) {
+    $this->setTranslator($job->getTranslator());
+
+    try {
+      $job_id = $job->id();
+      $project_id = $this->newTranslationProject($job_id, $job_id, $job->getSetting('required_by'), $job->getSetting('quote_required'), $job->getSetting('category'), $job->getSetting('review'));
+      $job->addMessage('Created a new project in thebigword with the id: @id', ['@id' => $project_id], 'debug');
+
+      foreach ($job->getItems() as $job_item) {
+        /** @var RemoteMapping $remote_mapping */
+        $remote_mapping = \Drupal::entityTypeManager()
+          ->getStorage('tmgmt_remote')
+          ->create([
+            'tjid' => $job->id(),
+            'tjiid' => $job_item->id(),
+            'data_item_key' => 'tmgmt_thebigword',
+            'remote_identifier_1' => 'ProjectId',
```

Comment:
as I said above. providing a fixed remote identifier doesn't make sense. just define that identifier_1 is the project_id.